src/hotspot/share/memory/arena.hpp

```diff
@@ -101,20 +101,17 @@ class Arena : public CHeapObj<mtNone> {
   size_t _size_in_bytes;        // Size of arena (used for native memory tracking)
 
   debug_only(void* malloc(size_t size);)
-  debug_only(void* internal_malloc_4(size_t x);)
 
-  void signal_out_of_memory(size_t request, const char* whence) const;
-
-  bool check_for_overflow(size_t request, const char* whence,
-      AllocFailType alloc_failmode = AllocFailStrategy::EXIT_OOM) const {
-    if (UINTPTR_MAX - request < (uintptr_t)_hwm) {
-      if (alloc_failmode == AllocFailStrategy::RETURN_NULL) {
-        return false;
-      }
-      signal_out_of_memory(request, whence);
+  void* internal_malloc_words(size_t x, AllocFailType alloc_failmode = AllocFailStrategy::EXIT_OOM)  {
```

Comment:
If UseMallocOnly is retained, I think a better name for this would be internal_amalloc_only, to avoid confusion with the case of actually using malloc.  (And yes, this can be taken as another argument for nuking that option.)

Reply:
Ok.  I'll file another RFE for removing UseMallocOnly.

Reply:
I decided that UseMallocOnly found an interesting bug once and there's a test for it, so it might find another interesting bug someday, so I added a comment and decided to leave it.  We can decide some other time to remove it if we have a good alternative.